File: assets/rt/Var.java
package rt;

public class Var implements base._MagicVarImpl_1 {
  private volatile Object value;
  public Var(Object value) {
    this.value = value;
  }
  @Override public Object get$read() {
    return this.value;
  }
  @Override public Object get$mut() {
    return this.value;
  }
  @Override public Object swap$mut(Object x_m$) {
    var old = this.value;
    this.value = x_m$;
    return old;
  }
}

File: assets/rt-js/Var.js
export class rt$$Var {
  constructor(value) {
    this.value = value;
  }
  get$read$0() {
    return this.value;
  }
  get$mut$0() {
    return this.value;
  }
  swap$mut$1(newValue) {
    const old = this.value;
    this.value = newValue;
    return old;
  }
}
